fix: fixes @distube/ytdl-core is out of date and music not playing issue by switching to use yt-dlp to get streaming URL

tabloomoo opened the pull request:
Closes #1283

<!-- A brief description of changes -->
- Switched to `yt-dlp` to get the streaming URL, removing dependency on `@distube/ytdl-core`
- Added `docker-compose.example.yml` and updated documentation for running with docker-compose locally
- Deafens by default when joining voice channel

Mostly vibe-coded. I've tested the build and I'm able to run the bot and hear music. Feel free to make changes/abandon this PR as you see fit

- [x] I updated the changelog

SuperKali committed:
Bump: @distube/ytdl-core to v4.16.8

In the latest version, the issue has been resolved where starting an instance of ffmpeg caused the server to return a 403 Forbidden error (access denied), this preventing the playback of any music tracks.

JoeWHoward committed:
Release 2.11.1

shiftybitshiftr committed:
Update ytdl-core (#1261)

* Update package.json > @distube/ytdl-core to ^4.16.10

tabloomoo committed:
switched to yt-dlp, switched to use npm instead of yarn, by default self-deafens

tabloomoo committed:
revert back to yarn. updated dockerfiles, fixed lint issues, updated doc and package.json, added docker-compose

tabloomoo committed:
updated line ending

tabloomoo committed:
revert format changes

tabloomoo committed:
format changes

tabloomoo committed:
removed unnecessary logic

tabloomoo committed:
format

tabloomoo committed:
Merge branch 'master' into master

codetheweb commented:
#### :package: :robot: A new release has been made for this pull request.

To play around with this PR, pull `ghcr.io/museofficial/muse:pr-1284`.

Images are available for x86_64 and ARM64.

> Latest commit: 66fe0e07b3fb94d3b32708edf689773bece88a20
<!-- Sticky Pull Request Commentpr-release -->

tabloomoo committed:
restored yarn lock

tabloomoo committed:
Merge branch 'master' of github.com:tabloomoo/muse

tabloomoo committed:
regenerated yarn lock

Vinookieadv commented:
I just usually use the muse with VPN (Gluetun) for the connection, because my root server got somehow listed and needs to activate cookies.  
When i use this PR i get the message "ope: yt-dlp failed with code 1: ERROR: [youtube] ZAz3rnLGthg: Sign in to confirm you’re not a bot. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies. Also see  https://github.com/yt-dlp/yt-dlp/wiki/Extractors#exporting-youtube-cookies  for tips on effectively exporting YouTube cookies"

hideki2k02 commented:
Working as of 31 of October, time to play spooky songs 👻

Drahgoone reviewed (approved):
Tested and working as of 13Dec25.

Cosmo121 commented:
Working for me as well

quinnhuettner commented:
This looks to now be broken in my experience

tabloomoo committed:
fix(voice): upgrade @discordjs/voice to 0.19.1 for DAVE E2E encryption support

Discord now requires DAVE (Discord's A/V end-to-end encryption protocol)
on voice connections. @discordjs/voice 0.18.0 did not implement DAVE,
causing voice connections to cycle signalling->connecting indefinitely and
audio to autopause immediately after starting.

- Upgrade @discordjs/voice 0.18.0 -> 0.19.1
- Update discord-api-types 0.32.1 -> 0.38.41 (required peer dep)
- Update @types/ws to ^8.18.1
- Remove keepAlive Reflect.get workaround (broken in 0.18.0+)
- Document that Spotify integration requires Spotify Premium

tabloomoo commented:
> Is anyone having issues playing longer content, like +45 mins? I see the behavior of one second of audio and then nothing. Bot leaves channel. Nothing in logs.

Do you have repro steps? Like, what URL were you using?
